### DISCUSSION TITLE
get "undefined reference to `__roken_dbm_store'" error during make

### DISCUSSION THREAD

**Discussion by xfan2020** [heimdal/heimdal#795] [Q&A] [answered]
Hi team,
I tried to build heimdal 7.7.0 on Photon OS (https://github.com/vmware/photon), but got following error

```
make[2]: Entering directory '/home/xfan/heimdal-7.7.0/lib/otp'
  CCLD     otptest
/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_store'
/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_delete'
/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_fetch'
/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_close'
/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_open'
collect2: error: ld returned 1 exit status

```
Any help you can give will be appreciated!

**Comment by newdivide2014** [comment]
same error.
i build heimdal 7.7.0 on ubuntu 20.04.3  LTS, got same error.

```
/usr/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_store'
/usr/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_delete'
/usr/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_fetch'
/usr/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_close'
/usr/bin/ld: ./.libs/libotp.so: undefined reference to `__roken_dbm_open'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:692: otptest] Error 1
make[2]: Leaving directory '/home/bao/heimdal/lib/otp'
make[1]: *** [Makefile:588: all-recursive] Error 1
make[1]: Leaving directory '/home/bao/heimdal/lib'
make: *** [Makefile:632: all-recursive] Error 1
```
I need help.  plz~

**Comment by lhoward** [comment, marked as answer]
I've seen this on Ubuntu, I think you need to install `libdb` / development package.

**Reply by abartlet** [reply]
Yes, I installed random libdb dev packages until it worked.  I think it was the db6-dev one in the end, but I'm not on that computer right now.

**Reply by abartlet** [reply]
I'm not suggesting Heimdal has the resources to do this, but on Samba I ended up making configure fail early and suggest common package names for the things we needed to build, so that this would fail early.  We also have a really cool system of scripts to prepare all the packages we need called `bootstrap`, which replaced a quite fragile list of required packages in our wiki.

**Reply by newdivide2014** [reply]
> I've seen this on Ubuntu, I think you need to install libdb / development package.

Thanks a lot . That solves my problem
